llm_studio/src/utils/modeling_utils.py

```diff
@@ -1029,10 +1035,22 @@ def prepare_lora(cfg, backbone):
         bias="none",
         task_type="CAUSAL_LM",
     )
-    if cfg.architecture.gradient_checkpointing:
-        backbone.enable_input_require_grads()
+    # not needed anylonger with use_reentrant=False
+    # if cfg.architecture.gradient_checkpointing:
+    #     backbone.enable_input_require_grads()
+
     backbone = get_peft_model(backbone, lora_config)
 
+    for name, param in backbone.named_parameters():
+        # unfreeze base model's layers
+        if any(
+            unfreeze_layer in name
+            for unfreeze_layer in cfg.training.lora_unfreeze_layers
+        ):
+            if cfg.environment._local_rank == 0:
+                logger.info(f"Unfreezing layer: {name}")
+            param.requires_grad = True
```

Comment:
Same here, catch cases where LoRA is disabled.

Reply:
that one cant happen because it is in the lora path